server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -73,20 +73,32 @@ public long getTookMetric() {
 
     @Override
     protected void onPhaseStart(SearchPhaseContext context) {
-        phaseStatsMap.get(context.getCurrentPhase().getSearchPhaseName()).current.inc();
+        try {
+            phaseStatsMap.get(context.getCurrentPhase().getSearchPhaseName()).current.inc();
```

Comment:
I count 4 method calls in this line. The problem with this try-catch is that any one of them in theory could throw an IllegalArgumentException, and you probably don't want to swallow it in all cases. Also, exception handling in the control flow is generally something to avoid.

I might recommend revisiting that API on `SearchPhase` to make it explicit that it may not be possible to get a `SearchPhaseName`. Like if it returned an Optional you could do:

```
phase.tryGetSearchPhaseName().ifPresent(name -> {
    phaseStatsMap.get(name).current.inc();
});
```

Reply:
Makes sense. I've just pushed a commit changing `getSearchPhaseName` to return `Optional<SearchPhaseName>`.